hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientICMPManager.java

```diff
@@ -79,26 +77,25 @@ public ClientICMPManager(HazelcastProperties hazelcastProperties, ClientExecutio
         }
 
         if (icmpEnabled) {
-            echoFailFast(hazelcastProperties);
+            if (clientIcmpPingConfig.isEchoFailFastOnStartup()) {
+                echoFailFast();
+            }
             this.icmpFailureDetector = new PingFailureDetector<Connection>(icmpMaxAttempts);
         } else {
             this.icmpFailureDetector = null;
         }
 
     }
 
-    private void echoFailFast(HazelcastProperties hazelcastProperties) {
-        boolean icmpEchoFailFast = hazelcastProperties.getBoolean(ClientProperty.ICMP_ECHO_FAIL_FAST);
-        if (icmpEchoFailFast) {
-            logger.info("Checking that ICMP failure-detector is permitted. Attempting to create a raw-socket using JNI.");
+    private void echoFailFast() {
+        logger.info("Checking that ICMP failure-detector is permitted. Attempting to create a raw-socket using JNI.");
 
-            if (!ICMPHelper.isRawSocketPermitted()) {
-                throw new IllegalStateException("ICMP failure-detector can't be used in this environment. "
-                        + "Check Hazelcast Documentation Chapter on the Ping Failure Detector for supported platforms "
-                        + "and how to enable this capability for your operating system");
-            }
-            logger.info("ICMP failure-detector is supported, enabling.");
+        if (!ICMPHelper.isRawSocketPermitted()) {
+            throw new IllegalStateException("ICMP failure-detector can't be used in this environment. "
+                    + "Check Hazelcast Documentation Chapter on the Ping Failure Detector for supported platforms "
```

Comment:
I think there should be a client section in there, since the doc describes the member side using props.

Reply:
Yes, we will add documentation for client. 
Especially because configuration is all different now.